File: day:61/longestSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # base case
        if(len(nums) == 0): return 0;
        longest = 1;
    
        for num in nums:
            if(num - 1 not in nums):
                current = num;
                count = 1;
                while(current + 1 in nums):
                    count += 1;
                    current += 1;
                    longest = max(longest, count);
        
        return longest;

File: day:61/longestSequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    // base case
    if(nums.length === 0) return 0;
    let longest = 1;
    
    for(let num of nums) {
        if(!nums.includes(num - 1)) {
            let current = num;
        let count = 1;
        while(nums.includes(current + 1)) {
            count += 1;
            current += 1;
            longest = Math.max(longest, count);
        }
        }
        
    }
    return longest;
};
